plugin.py

```diff
@@ -304,7 +311,7 @@ def on_legend_click(self, current, column=0):
         if layer.providerType() == "spatialite":
             rev = 0
             try:
-                rev = versioning.revision( uri.database() )
+                rev = self.spversioning.revision( uri.database() )
```

Comment:
Instead of an empty ctor (above in the code) I think that it would be better to have a `self.versioning` member that is constructed here with connection info (and schema for pg).

After that, no need to distinguish between various case, we could use, e.g. `self.versioning.revision()` to get the revision, whatever the concrete type of `self.versioning` if it's not None.